app/src/main/java/com/github/se/signify/ui/screens/tutorial/TutorialScreen.kt

```diff
@@ -0,0 +1,144 @@
+package com.github.se.signify.ui.screens.tutorial
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.signify.R
+import com.github.se.signify.ui.navigation.NavigationActions
+import com.github.se.signify.ui.screens.home.HomeScreen
+import com.github.se.signify.ui.theme.Error
+import com.github.se.signify.ui.theme.OnBackground
+import com.github.se.signify.ui.theme.OnPrimary
+import com.github.se.signify.ui.theme.Primary
+
+@Composable
+fun TutorialScreen(navigationActions: NavigationActions, onFinish: () -> Unit) {
+  var step by remember { mutableIntStateOf(0) }
+
+  Box(modifier = Modifier.fillMaxSize()) {
+    // Display the HomeScreen in the background
+    HomeScreen(navigationActions)
+
+    // Overlay for tutorial steps
+    when (step) {
```

Comment:
## Requested changes

This is way too long of a switch. I would move the steps to some array, perhaps defined in a separate file. This would shorten the body of this `@Composable`, and make it easier to add steps in the future.

Reply:
https://github.com/orgs/Signify-epfl/projects/2/views/1?pane=issue&itemId=89923935